### ISSUE TITLE
webSocketEvent not called

### ISSUE THREAD

**Issue by Blast12345** [Links2004/arduinoWebSockets#527]
I have an ESP8266. I'm trying to have it connect to my server running on a Pi. 

Currently the server is implemented using the following code:
https://github.com/Blast12345/LightOrganServer/blob/master/src/main/kotlin/Server.kt

I slightly modified this client to simply print out the data it receives:
https://gist.github.com/Silverbaq/1fdaf8aee72b86b8c9e2bd47fd1976f4

When I run both - I see "WRITING DATA" printed repeatedly from the server terminal window.
I see "Fake Data!" printed repeatedly from the client terminal window.
So - I'm confident that the socket implementation works.

I started working with the ESP8266 after having confirmed that the server works.
I followed the example project in this repo, but trimmed the fat (so to speak) and filled in the missing details. Here is the code at the moment:
https://github.com/Blast12345/LightOrganClient/blob/master/LightOrganClient/LightOrganClient.ino

When I run the project - the client connects.
```
Connection established!
IP address:	192.168.4.7
```

I can see the server start sending it messages from the terminal.
But - webSocketEvent never is called. I never get any other output from the ESP8266.

I see output if I put a `println("ASDF")` in the loop() - so I know the ESP8266 isn't getting hung up.

I have NOT tried to form a socket connection between two ESP8266s (a client and a server) as I only have one ESP8266 at the moment. I ordered a few more, so I should be able to test by the weekend. Perhaps Java and this library handle sockets a little differently?

**TLDR**
I'm connecting to a server running on a Raspberry Pi. I can see the client connect, but the client's webSocketEvent event is never called.
Why might this be?

**Comment by Links2004** [comment]
is your Java code really a websocket server?
`java.net.ServerSocket` sounds more like plain TCP to me.

if you enable the debug on the websocket lib / ESP
it most likely shows you that your server is no HTTP / websocket server.

**Comment by Blast12345** [comment]
Ohhh that is a great point. I forgot that websockets and sockets are not the same.

For what it's worth - the output (with debug on) is:
```
scandone
Connecting to LightOrgan ...
1 2 scandone
state: 0 -> 2 (b0)
3 state: 2 -> 3 (0)
state: 3 -> 5 (10)
add 0
aid 1
cnt 
state: 5 -> 2 (2c0)
rm 0
4 reconnect
state: 2 -> 0 (0)
scandone
state: 0 -> 2 (b0)
state: 2 -> 3 (0)
state: 3 -> 5 (10)
add 0
aid 1
cnt 

connected with LightOrgan, channel 7
dhcp client start...
ip:192.168.4.7,mask:255.255.255.0,gw:192.168.4.1
5 

Connection established!
IP address:	192.168.4.7
```

I think I either need to add a proper websocket to my Kotlin code or implement a normal socket for my ESP8266. In any case - my issues seems to have not been with this library.

Thank you very much for your reply!
